File: results/RQ1/05_baseline_qualification/units/p11_custom_press_releases/java/replays/0001_seed_2810241787/p11_custom_press_releases_java/p11_custom_press_releases_java/src/main/java/Template.java
/**
 * An immutable template with deterministic {{key}} substitution.
 * The template contains a name and content where placeholders of the form {{key}}
 * are replaced with values from a provided map.
 */
public final class Template {
    private final String name;
    private final String content;

    /**
     * Constructs a new Template with the given name and content.
     *
     * @param name the name of the template
     * @param content the content of the template which may contain {{key}} placeholders
     */
    public Template(String name, String content) {
        this.name = name;
        this.content = content;
    }

    /**
     * Returns the name of the template.
     *
     * @return the name
     */
    public String getName() {
        return name;
    }

    /**
     * Returns the content of the template.
     *
     * @return the content
     */
    public String getContent() {
        return content;
    }

    /**
     * Applies custom fields to the template content by replacing {{key}} placeholders
     * with corresponding values from the provided map.
     * If a key is not present in the map, the placeholder remains unchanged.
     * If the fields map is null, it's treated as an empty map.
     *
     * @param fields the map of field names to values to substitute into the template
     * @return the content with placeholders replaced by values
     */
    public String applyCustomFields(java.util.Map<String, String> fields) {
        // Handle null fields map by treating it as empty
        if (fields == null) {
            fields = java.util.Collections.emptyMap();
        }

        String result = content;
        
        // Find all placeholders of the form {{key}} and replace them
        java.util.regex.Pattern pattern = java.util.regex.Pattern.compile("\\{\\{([^}]+)\\}\\}");
        java.util.regex.Matcher matcher = pattern.matcher(result);
        
        StringBuffer buffer = new StringBuffer();
        while (matcher.find()) {
            String key = matcher.group(1);
            String value = fields.get(key);
            // If the key is not found in the fields map, keep the placeholder unchanged
            // If the value is null, treat it as an empty string
            matcher.appendReplacement(buffer, value == null ? "" : java.util.regex.Matcher.quoteReplacement(value));
        }
        matcher.appendTail(buffer);
        
        return buffer.toString();
    }
}

File: results/RQ1/05_baseline_qualification/units/p11_custom_press_releases/python/replays/0001_seed_2037595148/p11_custom_press_releases_python/p11_custom_press_releases_python/press_releases/template.py
"""Template rendering for press releases."""

from dataclasses import dataclass
from typing import Dict, Optional


@dataclass(frozen=True)
class Template:
    """A press release template with content and rendering capabilities."""

    name: str
    content: str

    def render(self, fields: Optional[Dict[str, str]] = None) -> str:
        """Render the template with provided field values.

        Replaces placeholders in the template content with values from fields.
        Placeholders are in the form {{key}} and are replaced with the corresponding
        value from fields. If a field is not provided, the placeholder is left unchanged.
        Placeholders are replaced in sorted order of keys.

        Args:
            fields: Dictionary mapping placeholder keys to their replacement values.
                    If None, an empty dictionary is used.

        Returns:
            The rendered template string with placeholders replaced.
        """
        values = {} if fields is None else fields
        result = self.content
        for key in sorted(values):
            placeholder = "{{" + key + "}}"
            result = result.replace(placeholder, values[key])
        return result